## soname-strip.dpatch by Jens Peter Secher <jps@debian.org>
##
## Make it possible to override the stripping of the neko executable
## prior to building rest of compiler tools and introduce a proper
## SONAME for libneko.

Index: neko/Makefile
===================================================================
--- neko.orig/Makefile	2009-03-15 13:56:52.000000000 +0100
+++ neko/Makefile	2009-03-15 14:08:49.000000000 +0100
@@ -4,11 +4,13 @@
 
 CFLAGS = -Wall -O3 -fPIC -fomit-frame-pointer -I vm -D_GNU_SOURCE
 EXTFLAGS = -pthread
-MAKESO = $(CC) -shared -WBsymbolic
-LIBNEKO_NAME = libneko.so
+MAJOR=0
+MINOR=0
+MAKESO = $(CC) -shared -Wl,-soname,libneko.so.${MAJOR}
+LIBNEKO_NAME = libneko.so.${MAJOR}.${MINOR}
 LIBNEKO_LIBS = -ldl -lgc -lm
 NEKOVM_FLAGS = -Lbin -lneko
-STD_NDLL_FLAGS = ${NEKOVM_FLAGS}
+STD_NDLL_FLAGS = ${NEKOVM_FLAGS} -lm
 INSTALL_FLAGS =
 
 NEKO_EXEC = LD_LIBRARY_PATH=../bin:${LD_LIBRARY_PATH} NEKOPATH=../boot:../bin ../bin/neko
@@ -118,16 +120,19 @@
 
 bin/${LIBNEKO_NAME}: ${LIBNEKO_OBJECTS}
 	${MAKESO} ${EXTFLAGS} -o $@ ${LIBNEKO_OBJECTS} ${LIBNEKO_LIBS}
+	ln -s ${LIBNEKO_NAME} bin/libneko.so
+	ln -s ${LIBNEKO_NAME} bin/libneko.so.${MAJOR}
 
 bin/neko: $(VM_OBJECTS)
-	${CC} ${CFLAGS} ${EXTFLAGS} -o $@ ${VM_OBJECTS} ${NEKOVM_FLAGS}
-	strip bin/neko
+	${CC} ${CFLAGS} -o $@ ${VM_OBJECTS} ${NEKOVM_FLAGS}
 
 bin/std.ndll: ${STD_OBJECTS}
-	${MAKESO} -o $@ ${STD_OBJECTS} ${STD_NDLL_FLAGS}
+	${MAKESO} ${EXTFLAGS} -o $@ ${STD_OBJECTS} ${STD_NDLL_FLAGS}
 
 clean:
 	rm -rf bin/${LIBNEKO_NAME} ${LIBNEKO_OBJECTS} ${VM_OBJECTS}
+	rm -rf bin/libneko.so bin/libneko.so.${MAJOR}
+	rm -rf bin/nekoml.std
 	rm -rf bin/neko bin/nekoc bin/nekoml bin/nekotools
 	rm -rf bin/std bin/*.ndll bin/*.n libs/*/*.o
 	rm -rf src/*.n src/neko/*.n src/nekoml/*.n src/tools/*.n
